core/App/ffi_operation.cpp

```diff
@@ -390,6 +390,298 @@ extern "C"
+
+        ehsm_keyspec_t keyspec = (ehsm_keyspec_t)payloadJson.readData_uint32("keyspec");
+        token_json = payloadJson.readData_JsonValue("importToken");
+
+        JsonObj token;
+        token.setJson(token_json);
+
+        importToken = (ehsm_data_t *)malloc(APPEND_SIZE_TO_DATA_T(strlen(token.toString().c_str()) - 1));
+        importToken->datalen = APPEND_SIZE_TO_DATA_T(token.toString().size() - 1);
+
+        memcpy_s(importToken->data, importToken->datalen, token.toString().c_str(), importToken->datalen);
+
+        memcpy_s(&cmk_tmp, sizeof(ehsm_keyblob_t), cmk, sizeof(ehsm_keyblob_t));
+
+        ret = GetParametersForImport(&cmk_tmp, keyspec, &pubkey_tmp);
+        if (ret != EH_OK)
+        {
+            retJsonObj.setCode(retJsonObj.CODE_FAILED);
+            retJsonObj.setMessage("Server exception.");
+            goto out;
+        }
+
+        if (pubkey_tmp.datalen == 0 || pubkey_tmp.datalen > UINT16_MAX)
+        {
+            retJsonObj.setCode(retJsonObj.CODE_FAILED);
+            retJsonObj.setMessage("Server exception.");
+            goto out;
+        }
+
+        cmk = (ehsm_keyblob_t *)malloc(APPEND_SIZE_TO_KEYBLOB_T(cmk_tmp.keybloblen));
+        if (cmk == NULL)
+        {
+            retJsonObj.setCode(retJsonObj.CODE_FAILED);
+            retJsonObj.setMessage("Server exception.");
+            goto out;
+        }
+
+        pubkey = (ehsm_data_t *)malloc(APPEND_SIZE_TO_DATA_T(pubkey_tmp.datalen));
+        if (pubkey == NULL)
+        {
+            retJsonObj.setCode(retJsonObj.CODE_FAILED);
+            retJsonObj.setMessage("Server exception.");
+            goto out;
+        }
+
+        pubkey->datalen = pubkey_tmp.datalen;
+        cmk->keybloblen = cmk_tmp.keybloblen;
+        cmk->metadata = cmk_tmp.metadata;
+
+        ret = GetParametersForImport(cmk, keyspec, pubkey);
+        if (ret != EH_OK)
+        {
+            retJsonObj.setCode(retJsonObj.CODE_FAILED);
+            retJsonObj.setMessage("Server exception.");
+            goto out;
+        }
+
+        publicKey = (char *)malloc(pubkey->datalen);
+        memcpy_s(publicKey, pubkey->datalen, pubkey->data, pubkey->datalen);
+        publicKey[pubkey->datalen] = '\0';
+        retJsonObj.addData_string("pubkey", publicKey);
+
+        // generate hmac space
+        hmac = (ehsm_data_t *)malloc(APPEND_SIZE_TO_DATA_T(EH_HMAC_SHA256_SIZE));
+        if (hmac == NULL)
+        {
+            retJsonObj.setCode(retJsonObj.CODE_FAILED);
+            retJsonObj.setMessage("Server exception.");
+            goto out;
+        }
+        hmac->datalen = EH_HMAC_SHA256_SIZE;
+
+        // generate a 32B cmk.
+        sessioncmk = (ehsm_keyblob_t *)malloc(APPEND_SIZE_TO_KEYBLOB_T(80));
+
+        sessioncmk->metadata.keyspec = EH_AES_GCM_256;
+        sessioncmk->metadata.origin = EH_INTERNAL_KEY;
+        sessioncmk->metadata.keyusage = EH_KEYUSAGE_ENCRYPT_DECRYPT;
+        sessioncmk->keybloblen = 80;
+
+        ret = CreateKey(sessioncmk);
```

Comment:
what's  difference about the sessioncmk and sessionkeyblob in above function? please unify the name

Reply:
fix